### DISCUSSION TITLE
How to prevent from from submission

### DISCUSSION THREAD

**Discussion by nezaboravi** [inertiajs/inertia#1226] [Help (Vue)]
Hello
My business logic requires to have a Submit button disabled unless all the form data (or at least required ones) are filled up.
Form is pretty long and growing.
Current state is here:
```
const form = useForm({
    user: {
        email: ref(null),
        name: ref(null),
    },
    company: {
        bankDetails: ref(null),
        address: ref(null),
        additionalInfo: ref(null)
    },
    billing: {
        name: ref(null),
        address: ref(null),
        email: ref(null)
    },
    invoice: {
        prefix: ref(null),
        number: ref(null),
        invoiceDate: ref(moment()),
        invoiceDueDate: ref(moment())
    }
})
```
Front end is Vue.
Backend is Laravel.

in the template part, i will have a button Submit, which shall be disabled by default.

How do i keep Submit button disabled until all the fields are properly filled up?

**Comment by madsh93** [comment]
This would be an issue to handle in Vue.

You could e.g. create a computed function that validates if all properties are != and return a Boolean. Then set the form submit button with

<button type="submit" :disabled="isFormValidated">Submit form</button>
